File: frontend/src/types/knowledge.ts
export interface KnowledgeCategoryCount {
  name: string;
  count: number;
  label: string;
}

export interface KnowledgeComponent {
  id: string;
  category: string;
  name: string;
  organization?: string;
  officialDocumentation?: string;
  githubRepository?: string;
  license?: string;
  priority?: string;
  lastVerified?: string;
  description?: string;
  keyFeatures: string[];
}

export interface KnowledgeRegistryResponse {
  totalComponents: number;
  lastSync: string;
  categories: KnowledgeCategoryCount[];
  components: KnowledgeComponent[];
  page: number;
  page_size: number;
  total_pages: number;
}

File: app/schemas/knowledge.py
from typing import List, Optional
from pydantic import BaseModel, Field

class KnowledgeCategoryCount(BaseModel):
    name: str
    count: int
    label: str

class KnowledgeComponent(BaseModel):
    id: str
    category: str
    name: str
    organization: Optional[str] = None
    officialDocumentation: Optional[str] = None
    githubRepository: Optional[str] = None
    license: Optional[str] = None
    priority: Optional[str] = None
    lastVerified: Optional[str] = None
    description: Optional[str] = None
    keyFeatures: List[str] = Field(default_factory=list)

class KnowledgeRegistryResponse(BaseModel):
    totalComponents: int
    lastSync: str
    categories: List[KnowledgeCategoryCount]
    components: List[KnowledgeComponent]
    page: int
    page_size: int
    total_pages: int
